front/src/common/authorization/components/RoleBasedContent.tsx

```diff
@@ -1,17 +1,45 @@
+import React from 'react';
+
+import type { BuiltinRole } from 'common/api/osrdEditoastApi';
+
 import useUserRoleCheck from '../hooks/useUserRoleCheck';
 
 export type RoleBasedContentProps = {
-  requiredRoles: string[];
+  requiredRoles: BuiltinRole[];
+  disableIfUnauthorized?: boolean;
   children: React.ReactNode;
 };
 
-const RoleBasedContent = ({ requiredRoles, children }: RoleBasedContentProps) => {
+const RoleBasedContent = ({
+  requiredRoles,
+  children,
+  disableIfUnauthorized = false,
+}: RoleBasedContentProps) => {
   const isRoleAllowed = useUserRoleCheck(requiredRoles);
 
   if (isRoleAllowed) {
     return children;
   }
 
+  if (disableIfUnauthorized) {
```

Comment:
Hm, that seems pretty specific to the cards on the homepage. Maybe it would be better to use `useUserRoleCheck()` inside `Home` and set a CSS class there on individual cards?

Reply:
It's also used in a few other places, but I'm a bit worried about having this kind of generic logic for pretty different use-cases…

Would suggest this

```ts
<div className="col-6 col-md-5 col-lg-4 col-xl-3" {...(operationalStudiesAllowed && { 'aria-disabled': true })}>
    ...
</div>
```

```css
.cardscontainer [aria-disabled] {
  pointer-events: none;
  opacity: 0.5;
}
```

and

```ts
<div role="button" {...(newProjectStudyScenarioAllowed && { disabled: true })}>
```

Reply:
Looks good!